Logging improvements to vUSB

pearsonk opened the pull request:
- Add logging to specific desired events
-- Device added
-- Device removed
-- Device assigned to VM
-- Device unassigned from VM
-- Policy rule added
-- Policy rule removed
-- Policy match for decision to allow assignment
-- Policy match for decision to deny assignment
-- Implicit policy deny due to no matching policy rule
- Move some low-level logs to DEBUG priority
- Move some error logs to ERROR priority

Signed-off-by: Kevin Pearson <kevin.pearson@ortmanconsulting.com>

Commit:
Logging improvements to vUSB

- Add logging to specific desired events
-- Device added
-- Device removed
-- Device assigned to VM
-- Device unassigned from VM
-- Policy rule added
-- Policy rule removed
-- Policy match for decision to allow assignment
-- Policy match for decision to deny assignment
-- Implicit policy deny due to no matching policy rule
- Move some low-level logs to DEBUG priority
- Move some error logs to ERROR priority

Signed-off-by: Kevin Pearson <kevin.pearson@ortmanconsulting.com>

dpsmith reviewed (approved):
Reviewed-by Daniel P. Smith <dpsmith@apertussolutions.com>

eric-ch commented:
Build [$10131](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10131)

eric-ch commented on src/policy.c:
```diff
@@ -522,6 +563,16 @@ policy_auto_assign_new_device(device_t *device)
+        device->serial,
+        vm->uuid,
+        vm->domid,
+        rule->pos);
```
SEGV here, rule can be `NULL`.

Triggers on [$10131](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10131) when plugging a USB thumbdrive when a VM is focused.
```
#0  policy_auto_assign_new_device (device=device@entry=0x428e80) at ../../git/src/policy.c:566
#1  0x0000000000406177 in udev_maybe_add_device (dev=dev@entry=0x42d420, auto_assign=auto_assign@entry=1) at ../../git/src/udev.c:494
#2  0x0000000000406775 in udev_event () at ../../git/src/udev.c:617
#3  0x0000000000403acf in main () at ../../git/src/main.c:135
p rule
$1 = (rule_t *) 0x0
```

`default_lookup` can fail and return `NULL`, maybe that's not intended?

pearsonk replied:
Ah, it's not that default_lookup is failing, it's that the policy lookup after doesn't track the rule used for making the decision. Added a commit to add a parameter to policy_is_allowed to be able to keep track of the rule used to make the policy decision. Built and tested that it addresses the vm-focus auto-assign.

eric-ch replied:
Ah I misread that.

`policy_is_allowed` now has a more complex behavior:
- Allow rule: Returns `true` and set `*rules_ptr` if `rules_ptr` is not `null`;
- Deny rule: Returns `false` and set `*rules_ptr` if `rules_ptr` is not `null`;
- Default Deny catch-all: Returns `false` without touching `*rules_ptr`;

Commit:
Add parameter to policy_is_allowed to track rule

Tracking the rule, when not passed as null pointer, can be used to
interrogate the rule used to make policy decision. Implicit deny will
still result in the rule pointer being a null pointer.

Signed-off-by: Kevin Pearson <kevin.pearson@ortmanconsulting.com>

eric-ch commented:
Build [$10132](http://openxt-builder.ainfosec.com:8010/#/builders/3/builds/10132)

eric-ch reviewed (approved):
LGTM.

eric-ch commented:
Will merge soon.
